src/modules/list.js

```diff
@@ -116,18 +114,56 @@ class List {
       const editInput = event.target.closest('.toDoTask');
       if (editInput && document.activeElement !== editInput) {
         const listItem = editInput.closest('.listItem');
-        const indexToEdit = parseInt(listItem.id, 10);
+        const indexToEdit = this.list.findIndex((task) => task.index === parseInt(listItem.id, 10));
         const updatedDescription = editInput.value;
-        const updatedTask = this.list.find(
-          (task) => task.index === indexToEdit,
-        );
+        const updatedTask = this.list[indexToEdit];
         updatedTask.description = updatedDescription;
         localStorage.setItem('list', JSON.stringify(this.list));
         this.showList();
       }
     });
   }
 
+  // Checkbox completed
+  checkBoxChanger(listTask, task) {
+    // Checkbox function to change the status of "completed"
+    const checkbox = listTask.querySelector('.checkBox');
+    const actualTask = listTask.querySelector('.toDoTask');
+    if (checkbox.checked) {
+      listTask.classList.add('checked');
+      actualTask.classList.add('taskChecked');
+    }
+    checkbox.addEventListener('click', () => {
+      task.completed = checkbox.checked;
+      if (checkbox.checked) {
```

Comment:
- [Optional] Good job adding the functionality for the checkbox @VelzckC0D3, but for the event, I suggest you use `change` instead of `click`. this is more suited for a checkbox element.